File: backend/TenderGo.Models/Requests/ResetPasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace TenderGo.Models.Requests

{
    public class ResetPasswordRequest
    {
        [Required(ErrorMessage = "Email address is required")]
        [EmailAddress(ErrorMessage = "Invalid email address format")]
        [RegularExpression(@"^[^@\s]+@[^@\s]+\.[^@\s]+$", ErrorMessage = "Email must be in a valid format (e.g. user@example.com)")]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string Token { get; set; }

        [Required]
        [MinLength(8)]
        public string NewPassword { get; set; }
    }
}

File: UI/lib/shared/models/requests/reset_password_request.dart
class ResetPasswordRequest {
  final String email;
  final String code;
  final String newPassword;

  ResetPasswordRequest({
    required this.email,
    required this.code,
    required this.newPassword,
  });

  Map<String, dynamic> toJson() {
    return {'email': email, 'code': code, 'newPassword': newPassword};
  }
}
